### PR TITLE
feat(netlify): added SWR implementation

### PR DESCRIPTION
**Pull request by Rigo-m**
### 🔗 Linked issue

resolves https://github.com/unjs/nitro/issues/1811

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

As stated in the issue, currently netlify "base" preset handles ISR by redirecting to either a normal function (no caching behaviour) or a builder function (cached on a per-URL basis). This approach has some downsides because of netlify ODB limitations (i.e.: no query parameters passed to the server, not so transparent cache control [specifically ttl parameter is quite hidden and not accessible directly])

This PR hopes to implement [new netlify caching swr strategy](https://www.netlify.com/blog/swr-and-fine-grained-cache-control/).

### A note on implementation:
I've arbitrarily implemented the SWR header when `isr` is set to true. This means that server response gets cached by up to 1 year in netlify edge cache servers, but gets revalidated each time. In contrast, setting a number in the `isr` routeRule defaults to the other caching method where page gets cached and not revalidated until cache expires.
Let me know if this behaviour is a good implementation of nitro routeRules API or needs to be changed.
### A note on documentation:
I'll document everything after changes gets approved

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Rigo-m** [comment]
As stated above, this is not yet tested but is a first implementation straight out of my head. Need help with testing properly in a netlify environment

**Comment by Rigo-m** [comment]
Was able to setup a test environment here https://nitro-deploy-swr-test.netlify.app/api/hello?querytest=test3 by forking nitro-deploys repository.

First attempt at implementing the feature was a failure, apparently Netlify disregard _headers file when setting cache control headers.

Manually using h3 `setHeader` inside of `defineEventHandler` worked.
I'm currently trying to set headers at the entry level, but for now unsuccessfully. Can you help me on that?

**Comment by Rigo-m** [comment]
I've added headers to the generated localCall response instead of appending them to the incoming request event.
This works on my tests, not sure if this is the correct implementation because I didn't quite understood how request headers gets transformed/used by nitro localCall handling.

Gonna fix test suite and mark this PR as ready for review

File: test/fixture/types.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { expectTypeOf } from "expect-type";
 import { describe, it } from "vitest";
 import { EventHandler, EventHandlerRequest, defineEventHandler } from "h3";
+// @ts-ignore
```
Comment (Rigo-m) [inline test/fixture/types.ts:4]: Tests failed on me here, added ts-ignore. Didn't know if I had to run some sort of `prepare` to fix this issue. Remove if unnecessary

**Comment by Rigo-m** [comment]
A note on `netlify-edge` preset: I've added the same headers to the response, did not test them for now. It's not clear from the article whether edge functions caching strategies works the same as in normal functions.

**Comment by Rigo-m** [comment]
ATM, Cache headers behaviour is not as predictable as ODB behaviour. Meaning it doesn't return stale content as you would expect.
Opened an issue (here)[https://github.com/netlify-labs/cache-control-and-swr/issues/1]
 and a support ticket (here)[https://answers.netlify.com/t/netlify-new-swr-cache-policy-behaviour-is-not-clear/104747]
 
I'll keep you up to date. In the meantime, I'd appreciate a review on the code written :)

**Comment by danielkellyio** [comment]
The reason the cache headers behavior isn't predictable seems to be based on different CDN nodes handling different requests:
https://answers.netlify.com/t/cannot-get-netlify-api-endpoint-caching-to-work-as-expected/105763/7

So I think the SWR support from Netlify isn't quite up to par yet.

**Comment by pi0** [comment]
Thanks for your work on this @Rigo-m ❤️ Moving to #2406

**Comment by Rigo-m** [comment]
Thanks for the heads up @pi0 💘

**Pull request closed without merging**